Mathlib/LinearAlgebra/GeneralLinearGroup.lean

```diff
@@ -64,6 +69,26 @@ theorem generalLinearEquiv_to_linearMap (f : GeneralLinearGroup R M) :
 theorem coeFn_generalLinearEquiv (f : GeneralLinearGroup R M) :
     (generalLinearEquiv R M f) = (f : M → M) := rfl
 
+section Functoriality
+
+variable {S N : Type*} [Semiring S] [AddCommMonoid N] [Module S N]
+  {σ : R →+* S} {σ' : S →+* R} [RingHomInvPair σ σ'] [RingHomInvPair σ' σ]
+
+/-- A semilinear equivalence from `V` to `W` determines an isomorphism of general linear
+groups. -/
+def compLinearEquiv (e : M ≃ₛₗ[σ] N) : GeneralLinearGroup R M ≃* GeneralLinearGroup S N :=
```

Comment:
```suggestion
def congrLinearEquiv (e : M ≃ₛₗ[σ] N) : GeneralLinearGroup R M ≃* GeneralLinearGroup S N :=
```
or perhaps `congr` or `LinearEquiv.generalLinearGroupCongr`, which works with dot notation as `e.generalLinearGroupCongr`

@YaelDillies probably has opinions as this is similar to their recently-contributed `MulEquiv.monoidHomCongrLeft` definitions.